core/src/main/java/org/elasticsearch/index/translog/TranslogWriter.java

```diff
@@ -90,6 +96,13 @@ private TranslogWriter(
         assert initialCheckpoint.maxSeqNo == SequenceNumbersService.NO_OPS_PERFORMED : initialCheckpoint.maxSeqNo;
         this.maxSeqNo = initialCheckpoint.maxSeqNo;
         this.globalCheckpointSupplier = globalCheckpointSupplier;
+        boolean assertionsEnabled = false;
```

Comment:
`assert (seenSequenceNumbers = new HashMap<>()) != null;` if you wanna have a one-liner?